pandas/core/algorithms.py

```diff
@@ -463,32 +468,117 @@ def _factorize_array(values, check_nulls, na_sentinel=-1, size_hint=None):
     return labels, uniques
 
 
-@deprecate_kwarg(old_arg_name='order', new_arg_name=None)
-def factorize(values, sort=False, order=None, na_sentinel=-1, size_hint=None):
-    """
-    Encode input values as an enumerated type or categorical variable
+_shared_docs['factorize'] = """
```

Comment:
If it is only a single entry, can we make this a variable (eg `_shared_docstring_factorize`) ?